### PR TITLE
Ory hydra charts synced with ory official

### PR DESCRIPTION
**Pull request by dariusztutaj**
Ory hydra charts synced with ory official

Repository used to sync with: https://github.com/ory/k8s/tree/v0.23.2

**Related issue(s)**
#14064 
#14230

### CODE REVIEW

File: resources/ory/charts/hydra/files/watch.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+set -eu
```
Comment (cnvergence) [inline resources/ory/charts/hydra/files/watch.sh:1]: I have removed this shell script with the sidecar container in this PR 
- https://github.com/kyma-project/kyma/pull/13705
We are controlling this logic in ory-reconciler

**Review by cnvergence — COMMENTED** [review]
Added quick comment

File: resources/ory/charts/hydra/templates/deployment.yaml [hunk 2]
```diff
@@ -40,75 +39,40 @@ spec:
...
         {{- end }}
-
-      {{- if .Values.deployment.serviceAccountName  }}
-      serviceAccountName: {{ .Values.deployment.serviceAccountName }}
-      {{- end }}
+      serviceAccountName: {{ include "hydra.serviceAccountName" . }}
+      automountServiceAccountToken: {{ .Values.deployment.automountServiceAccountToken }}
       containers:
+        {{- if .Values.watcher.enabled }}
+        - name: watcher
```
Comment (cnvergence) [inline resources/ory/charts/hydra/templates/deployment.yaml:53]: This is the sidecar container that it is not needed by us

**Comment by werdes72** [comment]
Works as expected, LGTM ✅

**Comment by dariusztutaj** [comment]
no more valid

**Pull request closed without merging**